src/main/java/net/fabricmc/loader/launch/knot/KnotClassLoader.java

```diff
@@ -144,7 +144,9 @@ public URL nextElement() {
 		synchronized (getClassLoadingLock(name)) {
 			Class<?> c = findLoadedClass(name);
 
-			if (c == null && !name.startsWith("com.google.gson.") && !name.startsWith("java.")) { // FIXME: remove the GSON exclusion once loader stops using it (or repackages it)
+			// FIXME: remove the GSON exclusion once loader stops using gson.
+			// We now repackage Gson's JsonReader so removal is now possible
+			if (c == null && !name.startsWith("com.google.gson.") && !name.startsWith("java.")) {
```

Comment:
Can this now be done? or do we need to repackage it in mixin? (Id like to repackage all of mixins libs, possibly appart from asm)

Reply:
I asked player about that. He wants to deal with the actual removal of gson and it's corresponding exemption a bit later.